Misc doc improvements (#400)
killbill/killbill-docs: userguide/tutorials/development.adoc

      </pluginExecution>
   </pluginExecutions>
</lifecycleMappingMetadata>
----

> . Go to `Window > Preferences > Maven > Lifecycle Mappings`. Click *Browse* and select the path of the `lifecycle-mapping-metadata.xml` file. Click `Apply and Close`.

Reviewer: ```suggestion
. Go to `Window > Preferences > Maven > Lifecycle Mappings`. Click *Browse* and select the path of the `lifecycle-mapping-metadata.xml` file. Click *Apply and Close*.
```

Reviewer: I haven't yet addressed this in our "style guide," but the standard I have been using is to bold face what gets clicked or selected. 

I am pretty confident about that one. What I'm not so confident about is when to use the code-type font (like Courier) in our in-line text. So far I've been using it for URLs, file names, settings, etc.

Author: Fixed here and in a couple of other places.

> . Build the project again (In Project Explorer, `Right click on killbill > Maven > Update Project > OK`).

Reviewer: ```suggestion
. Build the project again (In Project Explorer, right-click on *killbill > Maven > Update Project*, then select *OK*).
```

 tenants
 user_roles
 users
----
 
[[build]]
=== Build

Once you set up the code and database, the next step is to build the application. Kill Bill is a standard Maven project. In order to build Kill Bill, you need to follow the steps given below:

. Ensure that you have *Maven 3.5.2 or higher* (It can be downloaded from http://maven.apache.org/download.cgi[here]).

> . Maven requires JDK. Ensure that you have *JDK 8 or higher* (It can be downloaded from

Reviewer: ```suggestion
. Maven requires JDK. Ensure that you have *JDK 8 or higher*. (You can download it from
```

> https://www.oracle.com/in/java/technologies/javase-downloads.html[here]).

Reviewer: ```suggestion
https://www.oracle.com/in/java/technologies/javase-downloads.html[here].)
```

 # For PostgreSQL, use jdbc:postgresql://127.0.0.1:5432/killbill?currentSchema=killbillschema
 org.killbill.billing.osgi.dao.user=root
 org.killbill.billing.osgi.dao.password=root
----
+
. Naviagate to the *PROJECT_ROOT* directory.

. Start Kill Bill by running the following command (Replace *PROJECT_ROOT* with your actual project root):
+
[source,bash]
----
 mvn -Dorg.killbill.server.properties=file:///PROJECT_ROOT/profiles/killbill/src/main/resources/killbill-server.properties -Dlogback.configurationFile=./profiles/killbill/src/main/resources/logback.xml jetty:run
----
+
. This should display the following:
+
[source,bash]
----
 [INFO] Started ServerConnector@7de2f9a6{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
 [INFO] Started @95008ms
 [INFO] Started Jetty Server
----

. Open a browser window and type \http://localhost:8080. This should display the following page:
+

> image:https://github.com/killbill/killbill-docs/raw/v3/userguide/assets/img/development/killbill-homepage.png[align=center]

Reviewer: ```suggestion
image::https://github.com/killbill/killbill-docs/raw/v3/userguide/assets/img/development/killbill-homepage.png[align=center]
```

Reviewer: ^^If it's on its own line, it needs two colons after "image"

... Click on *External Location* and Navigate to your Maven local repository (`~/.m2/repository`). Select the jar file corresponding to the file that you want to debug. For the  `DefaultKillbillConfigSource` you need to select
`/org/kill-bill/billing/killbill-platform-base/0.40.4/killbill-platform-base-0.40.4-sources.jar` (At the time of writing, `0.40.4` is the latest version of this jar, however this can change so the exact version might be different for you)

.. Set up a breakpoint as required. For example, if it is an issue in starting the application, you would need to set up a breakpoint in  `DefaultKillbillConfigSource.java#L118`.

.. Click `Run > Debug Configurations`.

> .. Double click  `New Remote Java Application`.

Reviewer: ```suggestion
.. Double-click  `New Remote Java Application`.
```

 [ERROR]
 [ERROR] After correcting the problems, you can resume the build with the command
 [ERROR]   mvn <args> -rf :killbill-profiles-killbill
----

> This error is most probably because the some file is missing license information. You can build the application using the following command:

Reviewer: ```suggestion
This error is probably because some file is missing license information. You can build the application using the following command:
```
